js/web/lib/wasm/wasm-core-impl.ts

```diff
@@ -84,11 +84,14 @@ export const initRuntime = async(env: Env): Promise<void> => {
  * @param epName
  */
 export const initEp = async(env: Env, epName: string): Promise<void> => {
-  if (!BUILD_DEFS.DISABLE_WEBGPU && (epName === 'webgpu' || epName === 'webnn')) {
+  if (!BUILD_DEFS.DISABLE_WEBGPU && (epName === 'webgpu' || epName === 'webnn') && !env.webgpu.device) {
     // perform WebGPU availability check
     if (typeof navigator === 'undefined' || !navigator.gpu) {
       throw new Error('WebGPU is not supported in current environment');
     }
+    if (epName === 'webnn' && (typeof navigator === 'undefined' || !(navigator as any).ml)) {
```

Comment:
If epName is webnn, I'd hope all the logic related to WebGPU can be skipped. @fs-eire  is working on https://github.com/microsoft/onnxruntime/pull/19735, which might be a total solution to this.

Reply:
I agree. I checked his PR which doesn't touch the initialization of JSEP. @fs-eire, will you handle it as well? You know we are not familiar with JSEP, the logic related to WebGPU is combined quite closely with JSEP, it's complicated for us to separate them.

Reply:
Will think about this.

Reply:
Added support for WebNN in #19735

Reply:
Cool! Thanks much @fs-eire. @zesongw, let's close this PR.